test/reconnect_test.go

```diff
@@ -884,21 +884,218 @@ func TestAuthExpiredReconnect(t *testing.T) {
 	case <-time.After(2 * time.Second):
 		t.Fatal("Did not get the auth expired error")
 	}
-	select {
-	case s := <-stasusCh:
-		if s != nats.RECONNECTING {
-			t.Fatalf("Expected to be in reconnecting state after jwt expires, got %v", s)
+	WaitOnChannel(t, stasusCh, nats.RECONNECTING)
+	WaitOnChannel(t, stasusCh, nats.CONNECTED)
+	nc.Close()
+}
+
+func TestForceReconnect(t *testing.T) {
+	s := RunDefaultServer()
+
+	nc, err := nats.Connect(s.ClientURL(), nats.ReconnectWait(10*time.Second))
```

Comment:
Only one behavior that wanted to confirm is whether we get an error on disconnect or not, and it seems the disconnect callback is called but without an error which I think it is good, e.g.

```go
nats.DisconnectErrHandler(func(_ *nats.Conn, err error) {
	t.Logf("ERROR: %v", err) // error is nil here
}),
```

Reply:
Yes, the callback is invoked without error which I thought is a reasonable behavior.